src/libServer/LookupServer.cpp

```diff
@@ -896,42 +897,47 @@ string LookupServer::GetEthCall(const Json::Value& _json) {
     }
     code = contractAccount->GetCode();
   }
-  Address fromAddr;
+
   string result;
-  uint64_t amount{0};
-  uint64_t gasRemained =
-      2 * DS_MICROBLOCK_GAS_LIMIT;  // for now set total gas as twice the ds gas
-                                    // limit
   try {
+    Address fromAddr;
     if (_json.isMember("fromAddr")) {
       fromAddr = Address(_json["fromAddr"].asString());
     }
+
+    uint64_t amount{0};
     if (_json.isMember("amount")) {
       const auto amount_str = _json["amount"].asString();
       amount = strtoull(amount_str.c_str(), NULL, 0);
     }
+
```

Comment:
Why are we playing with gas here?

Reply:
the gas implementation was there already, I only moved the gasRemained closer to where it is used